Library/Homebrew/service.rb

```diff
@@ -0,0 +1,183 @@
+    def run_type(type = nil)
+      case T.unsafe(type)
+      when nil
+        @run_type
+      when "immediate", :immediate
+        @run_type = type.to_s
+      when RUN_TYPE_INTERVAL, RUN_TYPE_CRON
+        raise TypeError, "Service#run_type does not support timers"
+      when String
+        raise TypeError, "Service#run_type allows: '#{RUN_TYPE_IMMEDIATE}'/'#{RUN_TYPE_INTERVAL}'/'#{RUN_TYPE_CRON}'"
+      else
+        raise TypeError, "Service#run_type expects a string"
+      end
+    end
+
+    sig { params(variables: T.nilable(T::Hash[String, String])).returns(T.nilable(T::Hash[String, String])) }
+    def environment_variables(variables = {})
+      case T.unsafe(variables)
+      when nil
+        @environment_variables
+      when Hash
+        @environment_variables = variables
+      else
+        raise TypeError, "Service#environment_variables expects a hash"
+      end
+    end
+
+    # The directory where the formula's variable files should be installed.
+    # This directory is not inside the `HOMEBREW_CELLAR` so it persists
+    # across upgrades.
+    def var
+      HOMEBREW_PREFIX/"var"
+    end
+
+    # The directory where the formula's configuration files should be installed.
+    # Anything using `etc.install` will not overwrite other files on e.g. upgrades
+    # but will write a new file named `*.default`.
+    # This directory is not inside the `HOMEBREW_CELLAR` so it persists
+    # across upgrades.
+    def etc
+      HOMEBREW_PREFIX/"etc"
+    end
+
+    # The directory where the formula's binaries should be installed.
+    # This is symlinked into `HOMEBREW_PREFIX` after installation or with
+    # `brew link` for formulae that are not keg-only.
+    def opt_bin
+      opt_prefix/"bin"
+    end
+
+    # The directory where the formula's binaries should be installed.
+    # This is symlinked into `HOMEBREW_PREFIX` after installation or with
+    # `brew link` for formulae that are not keg-only.
+    def opt_sbin
+      opt_prefix/"sbin"
+    end
+
+    # A stable path for this formula, when installed. Contains the formula name
+    # but no version number. Only the active version will be linked here if
+    # multiple versions are installed.
+    def opt_prefix
+      HOMEBREW_PREFIX/"opt/#{@formula.name}"
+    end
+
+    def std_service_path_env
+      "#{HOMEBREW_PREFIX}/bin:#{HOMEBREW_PREFIX}/sbin:/usr/bin:/bin:/usr/sbin:/sbin"
+    end
+
+    # Returns a `String` plist.
+    # @return [String]
+    sig { returns(String) }
+    def to_plist
+      clean_command = []
+      @run.each do |item|
+        case item
+        when Pathname
+          clean_command << item.to_s
+        end
+      end
+
+      base = {
+        "Label"            => "homebrew.service.#{@formula.name}",
+        "RunAtLoad"        => @run_type == RUN_TYPE_IMMEDIATE,
+        "ProgramArguments" => clean_command,
+      }
+
+      base[:KeepAlive] = @keep_alive if @keep_alive == true
```

Comment:
Why is `KeepAlive` a symbol but `Label` is a string? I'd vote to make them all symbols.